### PR TITLE
Remove unneeded return statements

### PR DESCRIPTION
**Pull request by DanielFrantes**
Just some clippy warnings fixes.

### CODE REVIEW

File: library/core/src/str/pattern.rs [hunk 1]
```diff
@@ -1814,7 +1814,7 @@ fn simd_contains(needle: &str, haystack: &str) -> Option<bool> {
             }
             mask &= !(1 << trailing);
         }
-        return false;
```
Comment (compiler-errors) [inline library/core/src/str/pattern.rs:1817]: This is not correct, for the record.

**Review by compiler-errors — CHANGES_REQUESTED** [review]

**Comment by compiler-errors** [comment]
Also fix CI (by formatting the code correctly), and it'd probably be best to test the changes first locally.

**Comment by workingjubilee** [comment]
Have you read the contributing instructions? You seem to be manually fixing things that should be fixed via commands like `./x.py fmt`, and you mentioned "cargo", but cargo doesn't really work in this repo. Not in the usual way, anyways.

See https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html

File: library/core/src/str/pattern.rs (hunk 1 above)
Comment (danik292) [reply library/core/src/str/pattern.rs:1817]: Its now good?

**Comment by joboet** [comment]
Thank you! I'm surprised actually that rustc doesn't lint this...

@bors r+ rollup

**Comment by compiler-errors** [comment]
@bors r-

Please squash this!

**Comment by compiler-errors** [comment]
@joboet please make sure that changes like this are squashed in the future especially if they're cleanup prs like 7 lines long 😄

**Comment by danik292** [comment]
> @bors r-
> 
> Please squash this! You can use:
> 
> ```
> git reset HEAD~4
> git add library/
> git commit -m "commit message"
> ```

dont bors squash it automaticly?

**Comment by compiler-errors** [comment]
Nope

**Comment by danik292** [comment]
If I try to do what you suggest It dont work.

**Comment by joboet** [comment]
I mostly do
```sh
git rebase -i HEAD~4
```
and replace all `pick`s except the first one with `f` (fixup), which melds them into the previous commits.

**Comment by danik292** [comment]
what do you think by firs one?

The first or last one from chronological order?

**Comment by workingjubilee** [comment]
> what do you think by firs one?
> 
> The first or last one from chronological order?

if you run `git rebase -i HEAD~4` then git will run whatever you have set as your `EDITOR` environment variable and you will see the list of commits.

It will show you something like this:
```
pick 76b195675ea Update the unstable book
pick 65a9090aa3b Exclude just `std` from rustc deps
pick 1c6bffff9df Keep rustc's std copy
```
This is roughly chronological order.

You will want to change it so that it looks like this:
```
pick 76b195675ea Update the unstable book
squash 65a9090aa3b Exclude just `std` from rustc deps
squash 1c6bffff9df Keep rustc's std copy
```

**Comment by workingjubilee** [comment]
Anyway:
> If I try to do what you suggest It dont work.

You will probably need to provide, like, a *lot* more context than "it doesn't work".

**Comment by tgross35** [comment]
> You will want to change it so that it looks like this:
> 
> ```
> pick 76b195675ea Update the unstable book
> squash 65a9090aa3b Exclude just `std` from rustc deps
> squash 1c6bffff9df Keep rustc's std copy
> ```

(probably `f`/`fixup` rather than `s`/`squash` in this case, since the commits are just things like "fix test failures")

@danik292 just for reference, `s` appends the commit message to the previous commit's message, `f` just gets rid of the message. For things that fix up your own work, i.e. aren't relevant for the repo's history, use `f`.

**Comment by danik292** [comment]
I must add something that dont diserupt check-tidy and dont chenge code to code because git is rejecting my rebase because contained chenges is there.

sigh

Any idea what to do??

**Comment by tgross35** [comment]
> I must add something that dont diserupt check-tidy and dont chenge code to code because git is rejecting my rebase because contained chenges is there.
> 
> sigh
> 
> Any idea what to do??

Squashing shouldn't ever break tidy if it was already passing. Can you clarify what you are doing and what gets rejected? As in, actual commands / errors.

(you can also do `git rebase -i $(get merge-base master)` if you have pulled `master` again locally, for squashing without actually changing the base)

**Comment by workingjubilee** [comment]
> I must add something that dont diserupt check-tidy and dont chenge code to code because git is rejecting my rebase because contained chenges is there.

do you mean when you `git push`?

because that's intentional.
we're asking you to `git push --force` the squashed 4-into-1 commits

**Comment by danik292** [comment]
git push --force
Username for 'https://github.com': danik292
Password for 'https://danik292@github.com': 
Running pre-push script /home/daniel/rust/x test tidy
Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.04s
Building stage0 tool tidy (x86_64-unknown-linux-gnu)
    Finished `release` profile [optimized] target(s) in 0.14s
fmt check
fmt warning: Something went wrong running git commands:
fmt warning: rust-lang/rust remote not found
fmt warning: Falling back to formatting all files.
fmt: checked 5485 modified files
tidy check
x.py completions check
Build completed successfully in 0:00:05
Everything up-to-date






this is output of git push --force

**Comment by danik292** [comment]
I will close it as chenges dont worth stressing.

**Comment by tgross35** [comment]
Try pushing with `--no-verify` to skip local tidy and let CI handle it (you need to reopen before you push, otherwise GitHub complains)

**Pull request closed without merging**